WooCommerce/Classes/ViewModels/InAppPurchases/UpgradeViewState.swift

```diff
@@ -17,6 +17,23 @@ enum UpgradeViewState: Equatable {
             return false
         }
     }
+
+    var analyticsStep: WooAnalyticsEvent.InAppPurchases.Step? {
+        switch self {
+        case .loading, .purchasing:
+            return nil
+        case .loaded:
+            return .planDetails
+        case .waiting:
+            return .processing
+        case .completed:
+            return .completed
+        case .prePurchaseError:
+            return .prePurchaseError
+        case .purchaseUpgradeError:
+            return .purchaseUpgradeError
+        }
+    }
```

Comment:
I moved `analyticsStep` to be a property of `UpgradeViewState`, rather than an extension of it within  `UpgradeView`, but I have no strong opinion in keeping it as it was. My intention was to keep the view as "dumb" as possible.